[publicmain] iP

publicmain opened the pull request:
# NikoChatbot

> “Smart chatbots are transforming the way humans interact with technology.” – Anonymous

**NikoChatbot** 是一个功能强大的聊天机器人项目，旨在提供高效、智能的对话体验。它具有以下特点：

- **This chatbot is a master for managing your daily tasks!** crazy
- Easy to **extend** and **customize**

You Need：

1. [Download NikoChatbot here](https://github.com/publicmain/ip)
2. Run the program and start chatting

Here’s the main code to launch NikoChatbot:

```java
public static void main(String[] args) {
        Niko chatbot = new Niko("example.txt");
        chatbot.run();
    }
```
Feature List:

- [x] Able to manage multiple types of tasks

- [x] Able to read the tasks from a txt file 

- [x] Able to memorize the tasks from previous chat

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

publicmain committed:
Add fruits.txt

publicmain committed:
level 0 complete

publicmain committed:
Level 1 complete

publicmain committed:
level 4 complete

publicmain committed:
Revert "level 4 complete"

This reverts commit 2a94ae5ff2293a7d1f31c860d4eae03e3358a2f2.

publicmain committed:
level 3 complete

publicmain committed:
testing

publicmain committed:
level2 complete

publicmain committed:
level 3 complete

publicmain committed:
Level 4 complete

publicmain committed:
level 5 complete

publicmain committed:
level 6 complete

publicmain committed:
ENUM complete

publicmain committed:
final edition 1

publicmain committed:
final edition 2

publicmain committed:
complete

publicmain committed:
Level 7 complete

publicmain committed:
Level-8 complete

publicmain committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/Niko.java

publicmain committed:
A-MoreOOP

publicmain committed:
A-Packages

publicmain committed:
Merge commit 'f722dd13dd4650cefa798df7f9154592d81f753b' into add-gradle-support

publicmain committed:
no message

publicmain committed:
no message

publicmain committed:
Merge branch 'add-gradle-support'

publicmain committed:
A-Gradle

publicmain committed:
A-JUnit

publicmain committed:
A-JavaDoc

publicmain committed:
code standrad

publicmain committed:
level-9

publicmain committed:
level-9 complete

publicmain committed:
Merge branch 'branch-A-CodingStandard'

publicmain committed:
Merge branch 'branch-A-JavaDoc' into branch-Level-9

# Conflicts:
#	example.txt
#	src/main/java/Niko/Task/TaskList.java

JJtan2002 commented on src/main/java/Niko/Command/FindCommand.java:
```diff
@@ -0,0 +1,52 @@
+            }
+
+        } catch (Exception e) {
+            // 捕获异常并显示错误信息
```
Should this be in English?

JJtan2002 commented on src/main/java/Test.java:
```diff
@@ -0,0 +1,29 @@
+
+
+        String formattedDate = convertDate(inputDate);
+        System.out.println(formattedDate);  // 输出: 2019-12-02
```
Should this be in English?

JJtan2002 commented on src/main/java/Niko/Task/Event.java:
```diff
@@ -0,0 +1,49 @@
+
+    @Override
+    public String toString() {
+        return "[E]" + super.toString() + " (from: " + start.format(DateTimeFormatter.ofPattern("MMM d yyyy h:mm a", Locale.ENGLISH))
```
Line should not exceed 120 characters?

JJtan2002 commented on src/main/java/Niko/Command/AddCommand.java:
```diff
@@ -0,0 +1,51 @@
+//
+// Source code recreated from a .class file by IntelliJ IDEA
+// (powered by FernFlower decompiler)
+//
+
```
Should this comment be removed?

JJtan2002 reviewed (commented):
Looks good to me! Just some minor changes to make. Good job!

publicmain committed:
Fix Code Style

publicmain committed:
GUI

publicmain committed:
no message

voldiemort commented on src/main/java/niko/command/AddCommand.java:
```diff
@@ -0,0 +1,52 @@
+     * @param tasks   The task list to which the task will be added.
+     * @param ui      The UI to be updated with the new task.
+     * @param storage The storage where the updated task list will be written.
+     * @return
```
Might be good to either add content to the @return field or omit the line if it's self-explanatory.

voldiemort commented on src/main/java/niko/command/Command.java:
```diff
@@ -0,0 +1,34 @@
+     *
+     * @return {@code true} if this command exits the application; {@code false} otherwise.
+     */
+    public boolean isExit() {
```
Should this be represented as an attribute of the class with a getter method rather than just a method? Would save the hassle of having to override the method in subclasses whenever you want `isExit = true`.

voldiemort commented on src/main/java/niko/command/FindCommand.java:
```diff
@@ -0,0 +1,56 @@
+     * @param tasks   The task list to search within.
+     * @param ui      The user interface to display messages.
+     * @param storage The storage to save data if needed.
+     * @return
```
Same comment as above on the empty @return line. Noticed in some other parts of the code as well.

voldiemort commented on src/main/java/niko/task/Event.java:
```diff
@@ -0,0 +1,55 @@
+            this.start = DateTimeParser.parseDateTime(start);
+            this.end = DateTimeParser.parseDateTime(end);
+        } catch (DateTimeParseException ignore) {
+            // Intentionally left blank
```
I like that you comment where areas are intentionally left blank.

voldiemort commented on src/test/java/niko/command/AddCommandTest.java:
```diff
@@ -0,0 +1,34 @@
+import niko.task.TaskList;
+import niko.task.Todo;
+
+
```
Might be good to cleanup areas with excessive number of empty lines. Noticed some of these in other areas of the code as well.

voldiemort reviewed (commented):
Looks good overall!
I like how the classes are split into packages.

publicmain committed:
A-Assertions

publicmain committed:
Revert "no message"

This reverts commit ae729b60f20540775b32d87d55909346b02d0e77.

publicmain committed:
A-CodeQuality

publicmain committed:
A-Streams

publicmain committed:
Merge pull request #1 from publicmain/branch-A-Assertions

A-Assertions

publicmain committed:
Merge branch 'master' into branch-A-CodeQuality

publicmain committed:
Merge pull request #2 from publicmain/branch-A-CodeQuality

Branch a code quality

publicmain committed:
Merge branch 'master' into branch-A-Streams

publicmain committed:
Merge pull request #3 from publicmain/branch-A-Streams

A-Streams

publicmain committed:
no message

publicmain committed:
Merge branch 'branch-A-Assertions'

# Conflicts:
#	example.txt

publicmain committed:
Merge branch 'branch-A-CodeQuality'

# Conflicts:
#	example.txt
#	src/main/java/niko/main/DialogBox.java

publicmain committed:
Update MainWindow.java

publicmain committed:
Merge branch 'master' of https://github.com/publicmain/ip

# Conflicts:
#	example.txt
#	src/main/java/niko/command/ListCommand.java

publicmain committed:
Merge branch 'master' of https://github.com/publicmain/ip

# Conflicts:
#	example.txt
#	src/main/java/niko/command/ListCommand.java

publicmain committed:
Update DialogBox.java

publicmain committed:
Update Niko.java

publicmain committed:
Update Storage.java

publicmain committed:
Merge branch 'master' of https://github.com/publicmain/ip

# Conflicts:
#	example.txt
#	src/main/java/niko/command/ListCommand.java

publicmain committed:
no message

publicmain committed:
no message

publicmain committed:
The getUserDialog method previously duplicated the logic of creating a Label and ImageView. This redundancy could lead to potential inconsistencies in future updates.

This commit refactors getUserDialog to call getDukeDialog, eliminating code duplication and improving maintainability.

Let's refactor the getUserDialog method by:
* Reusing the getDukeDialog method to create the dialog box.
* Reducing the redundancy of Label and ImageView instantiation in getUserDialog.

This refactor simplifies the code and ensures consistent behavior between user and Duke dialogs.

publicmain committed:
Refactor getUserDialog method to reuse getDukeDialog

The getUserDialog method previously duplicated the logic of creating a Label and ImageView. This redundancy could lead to potential inconsistencies in future updates.

This commit refactors getUserDialog to call getDukeDialog, eliminating code duplication and improving maintainability.

Let's refactor the getUserDialog method by:
* Reusing the getDukeDialog method to create the dialog box.
* Reducing the redundancy of Label and ImageView instantiation in getUserDialog.

This refactor simplifies the code and ensures consistent behavior between user and Duke dialogs.

publicmain committed:
Enhance Parser error handling and input validation

The original Parser class lacked sufficient input validation, leading to potential crashes or unclear error messages when the user input was incomplete or improperly formatted.

This commit improves the error handling and input validation by:
* Adding trim() to handle extra whitespace in user input.
* Ensuring commands like "todo", "deadline", and "event" throw appropriate errors when the input is incomplete or improperly formatted.
* Introducing detailed error messages for commands like "delete", "mark", and "unmark" when task numbers are missing or invalid.
* Updating exception handling to provide more user-friendly feedback.

These changes improve user experience by providing clearer guidance when input errors occur and prevent potential runtime issues caused by incorrect input.

publicmain committed:
Refactor Parser class to extract command parsing methods

The original Parser class had all command parsing logic within a single method, which made it harder to maintain and extend.

This commit refactors the Parser class by:
* Extracting command-specific parsing logic (e.g., for todo, deadline, event, etc.) into dedicated private methods.
* Simplifying the main parse method, making it cleaner and easier to understand.
* Improving code modularity and maintainability by separating the concerns of each command parsing.

These changes make it easier to add or modify commands in the future, as the code is now more organized and extensible.

publicmain committed:
A-BetterGui

publicmain committed:
A-Personality

publicmain committed:
A-MoreErrorHandling

publicmain committed:
A-UserGuide

publicmain committed:
modification about readme v1

publicmain committed:
no message

publicmain committed:
Upload Ui.png

publicmain committed:
Upload Ui.png

publicmain committed:
Fix bugs and Update UG

publicmain committed:
Format all input/output in the same way

publicmain committed:
Reformat the UG

publicmain committed:
Update README

publicmain committed:
Update README
